queryparams/params.go

```diff
@@ -0,0 +1,239 @@
+	limit := defaultValue
+	asString := params.Get(Page)
+	if asString != "" {
+		pi, err := validator(Page, asString)
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": Page, "value": asString})
+			return 0, err
+		}
+		limit = pi
+	}
+
+	return limit, nil
+}
+
+func DatesFromParams(ctx context.Context, params url.Values) (string, string, error) {
+	var (
+		from, to         time.Time
+		fromDate, toDate string
+	)
+
+	yearString, monthString, dayString := params.Get(YearAfter), params.Get(MonthAfter), params.Get(DayAfter)
+	if yearString != "" && monthString != "" && dayString != "" {
+		year, err := yearValidator(YearAfter, yearString)
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": YearAfter, "value": yearString})
+			return "", "", err
+		}
+		month, err := monthValidator(MonthAfter, monthString)
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": MonthAfter, "value": monthString})
+			return "", "", err
+		}
+		day, err := dayValidator(DayAfter, dayString)
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": DayAfter, "value": dayString})
+			return "", "", err
+		}
+		from = time.Date(year, time.Month(month), day, 0, 0, 0, 0, time.UTC)
+		_, m, _ := from.Date()
+		if m != time.Month(month) {
+			log.Warn(ctx, "invalid day of month", log.Data{DayAfter: dayString, MonthAfter: monthString})
+			return "", "", fmt.Errorf("invalid day (%s) of month (%s)", dayString, monthString)
+		}
+		params.Set(YearAfter, strconv.Itoa(from.Year()))
+		params.Set(MonthAfter, strconv.Itoa(int(from.Month())))
+		params.Set(DayAfter, strconv.Itoa(from.Day()))
+		fromDate = from.Format(dateFormat)
+	}
+
+	yearString, monthString, dayString = params.Get(YearBefore), params.Get(MonthBefore), params.Get(DayBefore)
+	if yearString != "" && monthString != "" && dayString != "" {
+		year, err := yearValidator(YearBefore, yearString)
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": YearBefore, "value": yearString})
+			return "", "", err
+		}
+		month, err := monthValidator(MonthBefore, monthString)
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": MonthBefore, "value": monthString})
+			return "", "", err
+		}
+		day, err := dayValidator(DayBefore, dayString)
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": DayBefore, "value": dayString})
+			return "", "", err
+		}
+
+		to = time.Date(year, time.Month(month), day, 0, 0, 0, 0, time.UTC)
+		_, m, _ := to.Date()
+		if m != time.Month(month) {
+			log.Warn(ctx, "invalid day of month", log.Data{DayBefore: dayString, MonthBefore: monthString})
+			return "", "", fmt.Errorf("invalid day (%s) of month (%s)", dayString, monthString)
+		}
+		params.Set(YearBefore, strconv.Itoa(to.Year()))
+		params.Set(MonthBefore, strconv.Itoa(int(to.Month())))
+		params.Set(DayBefore, strconv.Itoa(to.Day()))
+		toDate = to.Format(dateFormat)
```

Comment:
could you let me know why the params are being set here ?

Reply:
Haha nice catch. This was a historical thing I forgot to tidy up - I was originally using the time.Date() feature normalising any values' outside their usual ranges', e.g. October 32 converts to November 1. So I was resetting the values in case this had happened. But I changed the implementation to give an error (after reading the ticket again), but forgot to remove the reset statements